test/fast/Sourcing nvm.sh should not modify parameters of caller

```diff
@@ -0,0 +1,5 @@
+#!/bin/sh
+
+set -- yes
```

Comment:
Right - but since `nvm.sh` is sourced with an argument directly, that's the use case I'm concerned about (see https://github.com/creationix/nvm/blob/master/test/sourcing/Sourcing%20nvm.sh%20with%20--install%20should%20install%20the%20default#L4-L17 for an example).

I'm not familiar with using `set` like this, so I'm not yet convinced it covers the same functionality.